src/elements/fields/PhoneField/index.tsx

```diff
@@ -0,0 +1,278 @@
+
+    const exampleNumber = exampleNumbers[curCountryCode];
+    setPlaceholder(
+      global.libphonenumber
+        .parsePhoneNumber(exampleNumber, curCountryCode)
+        .formatNational()
+    );
+  }, [curCountryCode, element]);
+
+  useEffect(() => {
+    const phoneCode = countryMap[curCountryCode].phoneCode;
+    const newNumber = `${phoneCode}${rawNumber}`;
+    if ((fullNumber || rawNumber) && newNumber !== fullNumber) {
+      setCurFullNumber(newNumber);
+      onChange(newNumber);
+    }
+  }, [onChange, triggerOnChange, curCountryCode]);
+
+  const supportedCountries = useMemo(
+    () =>
+      countryData
+        .filter(({ countryCode }) =>
+          global.libphonenumber.isSupportedCountry(countryCode)
+        )
+        .map(({ flag, countryCode, countryName, phoneCode }) => (
+          <li
+            key={countryCode}
+            css={{
+              padding: '5px 10px',
+              display: 'flex',
+              alignItems: 'center',
+              transition: '0.1s ease all',
+              '&:hover': {
+                backgroundColor: '#e6e6e6'
+              }
+            }}
+            onClick={() => {
+              setCurCountryCode(countryCode);
+              setRawNumber('');
+              setFormattedNumber('');
+              setShow(false);
+            }}
+          >
+            <span css={{ fontSize: '24px', marginRight: '7px' }}>{flag}</span>
+            {countryName}
+            {` (+${phoneCode})`}
+          </li>
+        )),
+    []
+  );
+
+  const servar = element.servar;
+  return (
+    <div
+      css={{
+        maxWidth: '100%',
+        position: 'relative',
+        pointerEvents: editable ? 'none' : 'auto',
+        ...applyStyles.getTarget('fc')
+      }}
+      {...elementProps}
+    >
+      {children}
+      {fieldLabel}
+      <div
+        css={{
+          position: 'relative',
+          width: '100%',
+          whiteSpace: 'nowrap',
+          overflowX: 'hidden',
+          ...applyStyles.getTarget('sub-fc')
+        }}
+      >
+        <div
+          css={{
+            position: 'absolute',
+            cursor: 'pointer',
+            top: '50%',
+            transform: 'translateY(-50%)',
+            left: '5px',
+            padding: '0 5px',
+            borderRadius: '4px',
+            transition: '0.2s ease all',
+            ...applyStyles.getTarget('fieldToggle'),
+            '&:hover': {
+              backgroundColor: '#e6e6e6'
+            }
+          }}
+          ref={triggerRef}
+          onClick={() => setShow(!show)}
+        >
+          {countryMap[curCountryCode].flag}
+        </div>
+        <Overlay
+          target={triggerRef.current}
+          show={show}
+          onHide={() => setShow(false)}
+          placement='bottom-start'
+        >
+          {(props) => {
+            ['placement', 'arrowProps', 'show', 'popper'].forEach(
+              (prop) => delete props[prop]
+            );
+            return (
```

Comment:
It would be really nice if this list had a search/filter on it.  It was slightly tedious to scroll looking for a country.  The filter could accept partial names or partial country codes.